modyn/tests/supervisor/internal/test_grpc_handler.py

```diff
@@ -325,34 +364,41 @@ def test_get_data_in_interval_batched(test_grpc_connection_established):
         )
 
 
+# test register pipeline at mocked metadata db
+# TODO(#302 remove): example from test_selector_manager.py
+# @patch("modyn.selector.internal.selector_manager.MetadataDatabaseConnection", MockDatabaseConnection)
+# @patch.object(SelectorManager, "_instantiate_strategy")
+# def test_register_pipeline(test__instantiate_strategy: MagicMock):
+#     with tempfile.TemporaryDirectory() as tmp_dir:
+#         config = get_modyn_config()
+#         config["selector"]["trigger_sample_directory"] = tmp_dir
+#         selec = SelectorManager(config)
+#         assert selec.register_pipeline(42, "{}") == 0
+@patch.object(GRPCHandler, "init_metadata_db", noop_init_metadata_db)
+@patch("modyn.supervisor.internal.grpc_handler.MetadataDatabaseConnection", MockDatabaseConnection)
 @patch("modyn.supervisor.internal.grpc_handler.grpc_connection_established", return_value=True)
-def test_register_pipeline_at_selector(test_grpc_connection_established):
+def test_register_pipeline(test_grpc_connection_established):
     mgr = enlighten.get_manager()
     pbar = mgr.status_bar(
         status_format="Test",
     )
 
     handler = GRPCHandler(get_simple_config(), mgr, pbar)
 
-    with patch.object(handler.selector, "register_pipeline") as mock:
-        mock.return_value = PipelineResponse(pipeline_id=42)
-
-        result = handler.register_pipeline_at_selector(
-            {"pipeline": {"name": "test"}, "training": {"dataloader_workers": 2, "selection_strategy": {}}}
-        )
-
-        assert result == 42
-        mock.assert_called_once_with(RegisterPipelineRequest(num_workers=2, selection_strategy=JsonString(value="{}")))
+    pipeline_config = get_minimal_pipeline_config()
+    pipeline_id = handler.register_pipeline(pipeline_config)
```

Comment:
I just found the mock DB connection that you moved here but I don't understand it really yet since it is not used :D Not 100% sure still why this works without a sqlite db or a mockdbconnection

Reply:
The mock DB connection is used when a test wants to use `MetadataDatabaseConnection`. I think that's why the register_pipeline test here and the old register_pipeline test in selector_manager works without a real DB. 
But, yeah, the test using mock DB is only testing that supervisor calls the db to register pipeline, not that the registration is reflected in the db.